src/System.IO.FileSystem/src/System/IO/Win32FileSystemEnumerable.cs

```diff
@@ -122,7 +123,7 @@ internal Win32FileSystemEnumerableIterator(string path, string originalUserPath,
             Debug.Assert(searchOption == SearchOption.AllDirectories || searchOption == SearchOption.TopDirectoryOnly);
             Debug.Assert(resultHandler != null);
 
-            _oldMode = Interop.Kernel32.SetErrorMode(Interop.Kernel32.SEM_FAILCRITICALERRORS);
+            _setBackOldMode = Interop.Kernel32.SetThreadErrorMode(Interop.Kernel32.SEM_FAILCRITICALERRORS, out _oldMode);
```

Comment:
This feels more and more like one of those things that we should drop in framework optimized for cloud and devices.

Would it better to set this just once process wide in the host if it actually matters, and drop it from the everywhere else?